### PR TITLE
[WIP] Replace deprecated functionality in “customer:hash:upgrade” command

### LINKED ISSUE #29611: [Issue] Replace deprecated functionality in “customer:hash:upgrade” command
This issue is automatically created based on existing pull request: magento/magento2#27333: magento/magento2#: Replace deprecated functionality in “customer:hash:upgrade” command [WIP]

---------

### Description (*)

PR replaces deprecated `customer->load()`, `customer->save()` functionality in the 

```bin/magento customer:hash:upgrade```

command.

Original PR: https://github.com/magento/magento2/pull/24243

CC: @dmytro-ch @ihor-sviziev 

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR replaces deprecated `customer->load()`, `customer->save()` functionality in the 

```bin/magento customer:hash:upgrade```

command.

Original PR: https://github.com/magento/magento2/pull/24243

CC: @dmytro-ch @ihor-sviziev 

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29611: magento/magento2#: Replace deprecated functionality in “customer:hash:upgrade” command [WIP]

### CODE REVIEW

**Comment by sidolov** [comment]
@magento create issue

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@sidolov @gabrieldagama could you add some info why it fails without build results?

File: app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php [hunk 1]
```diff
@@ -19,31 +19,29 @@
...
+     * UpgradeHashAlgorithmCommand constructor.
+     *
      * @param Encryptor $encryptor
+     * @param CustomerResourceModel $resourceModel
      */
     public function __construct(
-        CollectionFactory $customerCollectionFactory,
-        Encryptor $encryptor
+        Encryptor $encryptor,
+        CustomerResourceModel $resourceModel
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php:39]: Hi @atwixfirster,
Could you try to pass Proxy class for the resource model? I think it should resolve the issue.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@sidolov @gabrieldagama could you provide some data why tests are failing?

**Comment by sidolov** [comment]
@ihor-sviziev I see an error in log during install 
```
02:01:16 [Build Functional EE]   [Zend_Db_Statement_Exception (42)]                                           
02:01:16 [Build Functional EE]   SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento2_prod.ea  
02:01:16 [Build Functional EE]   v_entity_type' doesn't exist, query was: SELECT `main_table`.* FROM `eav_en  
02:01:16 [Build Functional EE]   tity_type` AS `main_table` 
```
Let's re-run tests to have new results because it looks not relevant to current PR.

**Comment by sidolov** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @atwixfirster,
Could you please review the following test failure https://github.com/magento/magento2/pull/27333#issuecomment-701473975 and update your PR accordingly ?

**Comment by atwixfirster** [comment]
> Hi @atwixfirster,
> Could you please review the following test failure [#27333 (comment)](https://github.com/magento/magento2/pull/27333#issuecomment-701473975) and update your PR accordingly ?

Hi, @ihor-sviziev !

I will try to fix but no promises.

Thank you!

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Hi @atwixfirster,
I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**